include/swift/SIL/SILInstruction.h

```diff
@@ -3075,16 +3075,45 @@ class SelectValueInst : public SelectInstBase<SelectValueInst, SILValue> {
 
 /// MetatypeInst - Represents the production of an instance of a given metatype
 /// named statically.
-class MetatypeInst : public SILInstruction {
+class MetatypeInst final
+    : public SILInstruction,
+      private llvm::TrailingObjects<MetatypeInst, Operand> {
```

Comment:
Nm. Read the trailing objects source in LLVM. It appears that this is the suggested way to implement this. Sorry for the noise.